src/main/java/pl/lodz/zzpj/kanbanboard/security/user/UserDetailsServiceImpl.java

```diff
@@ -0,0 +1,26 @@
+package pl.lodz.zzpj.kanbanboard.security.user;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import pl.lodz.zzpj.kanbanboard.entity.User;
+import pl.lodz.zzpj.kanbanboard.repository.UsersRepository;
+
+@Service
+public class UserDetailsServiceImpl implements UserDetailsService {
+	@Autowired
+	UsersRepository usersRepository;
+
+	@Override
+	@Transactional
+	public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+		User user = usersRepository.findUserByEmail(username)
+				.orElseThrow(() -> new UsernameNotFoundException("User Not Found with username: " + username));
```

Comment:
There is already a generic exception (NotFoundException) that should be thrown here.